### PR TITLE
Move GLOBAL_CACHE_FOLDER from atldld.base to atldld.constants

### LINKED ISSUE #5: Remove duplicate cache folder variable
The variables `atldld.base.GLOBAL_CACHE_FOLDER` and `atldld.utils.CACHE_FOLDER` are the same.

Maybe we can think about a new submodule `atldld.config` where we put constants like `GLOBAL_CACHE_FOLDER`.

Currently importing `GLOBAL_CACHE_FOLDER` from `atldld.base` leads to a delay due to its heavy imports. If we keep `atldld.config` light on imports then importing from it should be instant.

### PR DESCRIPTION
**Pull request by Stannislav**
Closes #5 

Having `GLOBAL_CACHE_FOLDER` in `atldld.base` is bad because importing that module is really slow.

I chose the name `atldld.constants` because I feel like we'll need to add more constants soon anyway. Open to discussions on that one.

Also removed all duplications of the cache folder and added tests.

### CODE REVIEW

**Review by jankrepl — COMMENTED** [review]
Looks great:)

One thing I am wondering about is how easy/difficult it is to patch these constants in tests?

I guess `monkeypatch.setattr` should be able to do this. Anyway, another approach would be to write a getter function e.g. `get_constant("GLOBAL_CACHE_FOLDER")` that will be used everywhere inside of the code and it can be patched easily.

The main motivation is to make sure that our tests are not writing to the actual global cache folder but a tempdir instead.

File: tests/test_constants.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from atldld.constants import GLOBAL_CACHE_FOLDER
+
+
+def test_global_cache_folder():
+    assert GLOBAL_CACHE_FOLDER.exists()
```
Comment (jankrepl) [inline tests/test_constants.py:21]: I am not a huge fan. I know that it is the current logic. However, a Python package should not test whether a user's filesystem contains a specific folder.

Maybe some patching here would be more appropriate

**Comment by Stannislav** [comment]
> Looks great:)
> 
> One thing I am wondering about is how easy/difficult it is to patch these constants in tests?
> 
> I guess `monkeypatch.setattr` should be able to do this. Anyway, another approach would be to write a getter function e.g. `get_constant("GLOBAL_CACHE_FOLDER")` that will be used everywhere inside of the code and it can be patched easily.
> 
> The main motivation is to make sure that our tests are not writing to the actual global cache folder but a tempdir instead.
> 
> Expanding on this idea, wrapping this inside of a function might be also helpful in the future if we introduce config files / environment variables through which the user can control this cache folder. All the logic could be implemented within the getter.

Yeah, all valid points!

So first of all, this is pretty much the functionality we already had (apart from creating the folder if it doesn't exist). I just wanted to move the constant to a different module + remove duplication. Maybe we can improve the actual way we do this in a new PR?

Patching: another way would be setting the env variable `HOME` to point to a temporary directory before loading this module. But I'm not sure I like this idea either.

But overall I agree 100% we can do better than hard-coding it into a constant :)

File: tests/test_constants.py (hunk 1 above)
Comment (Stannislav) [reply tests/test_constants.py:21]: Yeah, not a fan either 😆 like I said, maybe we can think about a better mechanism in a new issue. This PR was just to speed up the import  of `GLOBAL_CACHE_FOLDER` without changing the funtionality much.

We didn't use to create this folder, but I remember there was a bug in `atlalign` once that a function was trying to write to the cache folder that didn't exist, and I think we fixed it there by just creating a folder like this. I don't remember our code doing any checks on the existence of the cache folder, so I thought it was be safer that way. But we can drop it too if you guys think it's too much.

Like I said, I would suggest discussing better cache handling to a new issue/PR.

**Comment by jankrepl** [comment]
> > Looks great:)
> > One thing I am wondering about is how easy/difficult it is to patch these constants in tests?
> > I guess `monkeypatch.setattr` should be able to do this. Anyway, another approach would be to write a getter function e.g. `get_constant("GLOBAL_CACHE_FOLDER")` that will be used everywhere inside of the code and it can be patched easily.
> > The main motivation is to make sure that our tests are not writing to the actual global cache folder but a tempdir instead.
> > Expanding on this idea, wrapping this inside of a function might be also helpful in the future if we introduce config files / environment variables through which the user can control this cache folder. All the logic could be implemented within the getter.
> 
> Yeah, all valid points!
> 
> So first of all, this is pretty much the functionality we already had (apart from creating the folder if it doesn't exist). I just wanted to move the constant to a different module + remove duplication. Maybe we can improve the actual way we do this in a new PR?
> 
> Patching: another way would be setting the env variable `HOME` to point to a temporary directory before loading this module. But I'm not sure I like this idea either.
> 
> But overall I agree 100% we can do better than hard-coding it into a constant :)

You are right that we already had this functionality/issues before. Maybe we can create a new issue out of this and keep this PR the way it is?:)

File: tests/test_constants.py (hunk 1 above)
Comment (jankrepl) [reply tests/test_constants.py:21]: Let's keep things the way they are and let's create a new issue:)

Comment (jankrepl) [reply tests/test_constants.py:21]: (Conversation resolved - I do not have the button)

**Review by EmilieDel — APPROVED** [review]
LGTM! 

I agree with the other comments on this PR! :)

**Comment by Stannislav** [comment]
@jankrepl @EmilieDel I created a follow up issue: #58 

Feel free to add more ideas :)

**Review by jankrepl — APPROVED** [review]
Looks great! Thank you!

**Pull request merged**